controller/proxy-injector/webhook.go

```diff
@@ -72,38 +73,40 @@ func Inject(api *k8s.API,
 		} else {
 			parent = &objs[0]
 		}
-	}
+		proxyInjectionAdmissionRequests.With(admissionRequestLabels(strings.ToLower(ownerRef.Kind), request.Namespace)).Inc()
```

Comment:
it looks like this got restructured slightly to move more code inside the `ownerRef != nil` if block.  What's the reasoning behind that change?  Doesn't this make it so that bare pods with no owner will no longer get injected?

Reply:
The reasoning is that, we will use `ownerRef` during injection for both `request` and `response` metrics. So, we have to make sure its not null.

I did check the same for bare pods, the ownerRef is a `Pod` Kind in that case as per this code https://github.com/linkerd/linkerd2/blob/master/controller/k8s/api.go#L376 . Hence I was able to see the metric for pod too 
```
proxy_inject_admission_requests_total{namespace="mem-example",owner_kind="pod"} 1
proxy_inject_admission_responses_total{namespace="mem-example",owner_kind="pod",reason="injection_enable_annotation_absent",skip="true"} 1
```
 I don't see where `ownerRef` would be null though!

Reply:
ah, I see.  even though we never expect the `ownerRef` to be null, we could make this diff smaller and more targeted by just wrapping the stat increment lines in ownerRef null checks and leaving everything else alone.

e.g.
```
if ownerRef != nil {
    proxyInjectionAdmissionRequests.With(admissionRequestLabels(strings.ToLower(ownerRef.Kind), request.Namespace)).Inc()
}
....
if ownerRef != nil {
    proxyInjectionAdmissionResponses.With(admissionResponseLabels(strings.ToLower(ownerRef.Kind), request.Namespace, "true", reason)).Inc()
}
return admissionResponse, nil
```

Alternatively, we may want to simply omit the label if (for some reason) there is no ownerRef.

Reply:
Updated as per your suggestion.